src/devices/video/mc6847.cpp

```diff
@@ -559,11 +588,23 @@ void mc6847_base_device::device_config_complete()
 
 	if (!screen().refresh_attoseconds())
 	{
-		// FIXME: use correct raw parameters rather than this nonsense
-		screen().set_refresh_hz(m_tpfs > 310.0 ? 50 : 60);
-		screen().set_size(320, 243);
-		screen().set_visarea(0, 320-1, 1, 241-1);
-		screen().set_vblank_time(0);
+		// It is preferred to initialize the screen device using set_raw.
+		// Setting screen device's raw parameters requires values in terms of pixels.
+		// Multiplying the 6847's clock by 2 produces a pixel clock that allows
+		// 256 pixels of active video area per line, with apparently no roundoff
+		// error or drift.
+		//
+		// Note: This math is valid for NTSC.  If / when PAL is implemented, m_tpfs
+		// can be used to determine if this 6847 is PAL, and a different set_raw
+		// call can be made if helpful.
+		screen().set_raw(
+			(u32)(clock() * 2),
+			456,										// htotal
+			0,											// hbend
+			BMP_L_OR_R_BORDER * 2 + BMP_ACTIVE_VIDEO,	// hbstart
+			m_tpfs,										// vtotal
+			0,											// vbend
+			LINES_UNTIL_RETRACE);						// vbstart
```

Comment:
The rest of this file doesn’t use the abbreviated integer types.  Please follow the pattern.

Reply:
Indeed!  Changed.